### PR TITLE
bugfix: Support serialization for dm.jdbc.driver.DmdbTimestamp (#6701)

### LINKED ISSUE #6701: 集成了达梦数据库，数据的前镜像中有个字段是dm.jdbc.driver.DmdbTimestamp，但是在回滚反序列化时报错了
报错信息如下：
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `dm.jdbc.driver.DmdbTimestamp` (although at least one Creator exists): no long/Long-argument constructor/factory method to deserialize from Number value (1721985847000)

### PR DESCRIPTION
**Pull request by lyl2008dsg**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did
This pull request introduces support for serialization and deserialization of the dm.jdbc.driver.DmdbTimestamp class. The changes ensure that environments lacking the DM JDBC driver do not encounter errors or unnecessary logging.

Changes Made

1. Conditional Registration of Serializers and Deserializers:
* Added logic to register serializers and deserializers for dm.jdbc.driver.DmdbTimestamp only if the class is present in the classpath.
* This avoids issues in environments where the DM JDBC driver is not available.

2. Error Handling:
* No error logs are recorded if the dm.jdbc.driver.DmdbTimestamp class is not found.
* This is to prevent confusion for users who do not have the DM JDBC driver installed.

### Ⅱ. Does this pull request fix one issue?

fixes https://github.com/apache/incubator-seata/issues/6701

### CODE REVIEW

**Review by funky-eyes — COMMENTED** [review]
我认为你应该需要在rm-datasource中引入以下依赖
I think you should need to introduce the following dependencies in rm-datasource
```
            <dependency>
                <groupId>com.dameng</groupId>
                <artifactId>DmJdbcDriver18</artifactId>
                <scope>test</scope>
            </dependency>
```

**Comment by lyl2008dsg** [comment]
> 我认为你应该需要在rm-datasource中引入以下依赖 I think you should need to introduce the following dependencies in rm-datasource
> 
> ```
>             <dependency>
>                 <groupId>com.dameng</groupId>
>                 <artifactId>DmJdbcDriver18</artifactId>
>                 <scope>test</scope>
>             </dependency>
> ```
done.
Yes, I think so too. It can improve unit test coverage.

File: rm-datasource/src/test/java/org/apache/seata/rm/datasource/undo/parser/JacksonUndoLogParserTest.java [hunk 1]
```diff
@@ -99,6 +106,48 @@ public void encode() throws NoSuchFieldException, IllegalAccessException, IOExce
         bytes = mapper.writeValueAsBytes(field);
         sameField = mapper.readValue(bytes, Field.class);
         Assertions.assertTrue(DataCompareUtils.isFieldEquals(field, sameField).getResult());
+
+    }
+
+    @Test
+    public void testSerializeAndDeserializeDmdbTimestamp() throws NoSuchFieldException, IllegalAccessException, ClassNotFoundException, NoSuchMethodException, InvocationTargetException, IOException {
+        Assumptions.assumeTrue(checkClassExists("dm.jdbc.driver.DmdbTimestamp"), "DmdbTimestamp class not found, skipping tests.");
```
Comment (funky-eyes) [inline rm-datasource/src/test/java/org/apache/seata/rm/datasource/undo/parser/JacksonUndoLogParserTest.java]:
```suggestion
```

Comment (lyl2008dsg) [reply rm-datasource/src/test/java/org/apache/seata/rm/datasource/undo/parser/JacksonUndoLogParserTest.java]: done

**Review by funky-eyes — APPROVED** [review]
LGTM

**Review by xingfudeshi — APPROVED** [review]
LGTM

**Pull request merged**